File: frontend/src/api/onboarding.ts
import { apiGet, apiPost } from './client';

export interface QuestionOption {
  id: string;
  label: string;
}

export interface Question {
  id: string;
  dimension_hint?: string;
  type: 'single' | 'multi' | 'open';
  prompt: string;
  options?: QuestionOption[];
  placeholder?: string;
}

export interface OnboardingAnswer {
  question_id: string;
  choice?: string | string[];
  free_text?: string;
}

export interface OnboardingSubmitResponse {
  dimensions: Record<string, number>;
  reasoning: string;
  snapshot_id: number;
}

export async function fetchOnboardingQuestions(): Promise<Question[]> {
  const res = await apiGet<{ questions: Question[] }>('/api/onboarding/questions');
  return res.questions;
}

export async function submitOnboarding(
  studentId: string,
  answers: OnboardingAnswer[]
): Promise<OnboardingSubmitResponse> {
  return apiPost<OnboardingSubmitResponse>('/api/onboarding/submit', {
    student_id: studentId,
    answers,
  });
}

File: backend/src/selflearn/schemas/onboarding.py
"""Onboarding 路由请求/响应 schema。"""
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class OnboardingAnswer(BaseModel):
    question_id: str
    choice: str | list[str] | None = None
    free_text: str | None = None


class OnboardingSubmitRequest(BaseModel):
    student_id: str
    answers: list[OnboardingAnswer] = Field(..., min_length=1)


class OnboardingSubmitResponse(BaseModel):
    dimensions: dict[str, float]
    reasoning: str
    snapshot_id: int


class OnboardingQuestionsResponse(BaseModel):
    questions: list[dict[str, Any]]
